addons/website_slides_survey/controllers/slides.py

```diff
@@ -3,52 +3,33 @@
 
 from odoo.addons.website_slides.controllers.main import WebsiteSlides
 
+from odoo import http
 from odoo.http import request
 
+import werkzeug.urls
+import werkzeug.utils
+from werkzeug.exceptions import BadRequest
+
 
 class WebsiteSlides(WebsiteSlides):
 
+    @http.route(['/slides_survey/certification_url/get'], type='http', auth='user', website=True)
+    def slide_certification_get_url(self, slide_id):
+        fetch_res = super(WebsiteSlides, self)._fetch_slide(slide_id)
+        if fetch_res.get('error'):
+            return BadRequest()
```

Comment:
Looks like BadRequest is an exception so you should `raise` it instead of returning it.

Reply:
Done